lib/buildlog/mod.rs

```diff
@@ -45,6 +106,37 @@ impl BuildLog {
 		BuildLog::read_from(file)
 	}
 
+	pub fn write(&self, file: impl AsRef<Path>) -> Result<(), Error> {
+		// TODO: should we append to this file instead of truncating it?
+		self.write_to(File::create(file)?)
+	}
+
+	pub fn write_to(&self, file: File) -> Result<(), Error> {
+		let mut file = BufWriter::new(file);
+
+		file.write_all(b"# ninja log v5\n")?;
+
+		// Write entries in order of finishing time. Note that this BuildLog does not
+		// contain any 'dead' entries like Ninja does, so writing it to file removes all
+		// 'dead' entries immediately, while ninja would only do that later.
```

Comment:
When you run `ninj b`, we still do read everything from `a` into the log, but it may indeed become outdated. How are outdated entries marked in the build log?

Reply:
I thought you wrote a new log every time, only including the entries that you added in the current run. But it still includes everything that was already in the log and was not touched, right? In that case, never mind.